### PR TITLE
Add continuous testing to helm chart

### PR DESCRIPTION
**Pull request by krajorama**
Support different kind of authentication methods.

This is for better testing of #2778 and also parity with jsonnet which already has continuous test support.

Signed-off-by: György Krajcsovits <gyorgy.krajcsovits@grafana.com>

### CODE REVIEW

**Comment by krajorama** [comment]
I've tested it against mimir, values:
```
continuous_test:
   enabled: true
```

And GEM after creating the mimir-continuous-test tenant and an access token which I placed in a secret:
```
continuous_test:
  enabled: true
  extraEnvFrom:
    - secretRef:
        name: tenant-password
  auth:
    type: basicAuth
    password: '$(TEST_ACCESS_TOKEN)'
```

**Comment by dimitarvdimitrov** [comment]
these are my suggestions. It's easier if I give them as a patch, since I applied them and they work locally. Most are nitpicks. One notable one is updating the image in the weekly releases


```patch
Index: operations/helm/charts/mimir-distributed/values.yaml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/operations/helm/charts/mimir-distributed/values.yaml b/operations/helm/charts/mimir-distributed/values.yaml
--- a/operations/helm/charts/mimir-distributed/values.yaml	(revision a2f6a89b43ca95bb67c1881b90f31c0a2cd3139b)
+++ b/operations/helm/charts/mimir-distributed/values.yaml	(date 1664960497348)
@@ -2497,7 +2497,10 @@
     annotations: {}
     labels: {}
 
-# Settings for the smoke-test job.
+# -- Settings for the smoke-test job. This is meant to run as a Helm test hook
+# (`helm test RELEASE`) after installing the chart. It quickly verifies
+# that writing and reading metrics works. Currently not supported for
+# installations using GEM token-based authentication.
 smoke_test:
   image:
     repository: grafana/mimir-continuous-test
@@ -2512,26 +2515,30 @@
   # -- The name of the PriorityClass for smoke-test pods
   priorityClassName: null
 
-# Settings for continous testing
+# -- Settings for mimir-continuous-test.
+# This continuously writes and reads metrics from Mimir.
+# https://grafana.com/docs/mimir/latest/operators-guide/tools/mimir-continuous-test/
 continuous_test:
   enabled: false
   # -- Number of replicas to start of continuous test
   replicas: 1
   image:
     repository: grafana/mimir-continuous-test
-    tag: r205-f2ad133
+    tag: r206-310ea58
     pullPolicy: IfNotPresent
+    # Note: optional pullSecrets are set in toplevel 'image' section, not here
+
   # -- Authentication settings of continuous test
   auth:
     # -- Type of authentication to use (tenantId, basicAuth, bearerToken)
     type: tenantId
-    # -- TenantId for tenantId auth
+    # -- TenantId for tenantId auth. This is injected as the X-Scope-OrgID header on requests.
     tenantId: 'mimir-continuous-test'
     # -- Username for basicAuth auth
-    username: 'mimir-continuous-test'
-    # -- Password for basicAuth auth (note: can be environment variable from secret, e.g. $(PASSWORD))
+    username: null
+    # -- Password for basicAuth auth (note: can be environment variable from secret attached in extraEnvFrom, e.g. $(PASSWORD))
     password: null
-    # -- Bearer token for bearerToken auth (note: can be environment variable from secret, e.g. $(TOKEN))
+    # -- Bearer token for bearerToken auth (note: can be environment variable from secret attached in extraEnvFrom, e.g. $(TOKEN))
     bearerToken: null
   # -- The maximum number of series to write in a single request.
   numSeries: 1000
@@ -2540,22 +2547,24 @@
   # -- Interval between test runs
   runInterval: "5m"
 
-  # -- Annotations for the continuous_test Deployment
+  # -- Annotations for the continuous test Deployment
   annotations: {}
   # -- The SecurityContext for continuous test containers
   containerSecurityContext:
     readOnlyRootFilesystem: true
-  # Extra environment variables for continuous test containers
+  # -- Extra environment variables for continuous test containers
   env: []
-  # Extra command line arguments for the continuous test container
+  # -- Extra command line arguments for the continuous test container
   extraArgs: {}
-  # Extra environment from secret/configmap for continuous test containers
+  # -- Extra environment from secret/configmap for continuous test containers
   extraEnvFrom: []
-  # Extra volumes for the continuous test container
+  # -- Extra volumes for the continuous test container
   extraVolumes: []
-  # Extra volume mounts for the continuous test container
+  # -- Extra volume mounts for the continuous test container
   extraVolumeMounts: []
-  # Extra initcontainers for the continuous test Deployment
+  # -- Extra containers for the continuous test Deployment
+  extraContainers: []
+  # -- Extra initContainers for the continuous test Deployment
   initContainers: []
   # -- The name of the PriorityClass for continuous test pods
   priorityClassName: null
@@ -2566,15 +2575,20 @@
     requests:
       cpu: "1"
       memory: 512Mi
-  # Security context for the continuous test Deployment
+  # -- Security context for the continuous test Deployment
   securityContext: {}
-  # Service for monitoring continuous test
+  # -- Service for monitoring continuous test
   service:
     annotations: {}
     labels: {}
-  # Upgrade strategy for the continuous test Deployment
+  # -- Upgrade strategy for the continuous test Deployment
   strategy:
     type: RollingUpdate
     rollingUpdate:
       maxSurge: 0
       maxUnavailable: 1
+
+  nodeSelector: {}
+  affinity: {}
+  tolerations: []
+  terminationGracePeriodSeconds: 30
```

```patch
Index: .github/workflows/scripts/helm-weekly-release.sh
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.github/workflows/scripts/helm-weekly-release.sh b/.github/workflows/scripts/helm-weekly-release.sh
--- a/.github/workflows/scripts/helm-weekly-release.sh	(revision a2f6a89b43ca95bb67c1881b90f31c0a2cd3139b)
+++ b/.github/workflows/scripts/helm-weekly-release.sh	(date 1664959979816)
@@ -89,6 +89,7 @@
 update_yaml_node $values_file .image.tag $latest_mimir_tag
 update_yaml_node $values_file .smoke_test.image.tag $latest_mimir_tag
 update_yaml_node $values_file .enterprise.image.tag $latest_gem_tag
+update_yaml_node $values_file .continuous_test.image.tag $latest_mimir_tag
 update_yaml_node $chart_file .appVersion $(extract_r_version $latest_mimir_tag)
 update_yaml_node $chart_file .version $new_chart_version 
```

File: operations/helm/charts/mimir-distributed/values.yaml [hunk 1]
```diff
@@ -2511,3 +2514,84 @@ smoke_test:
...
+  # -- Authentication settings of continuous test
+  auth:
+    # -- Type of authentication to use (tenantId, basicAuth, bearerToken)
+    type: tenantId
+    # -- The tenant to use for tenantId or basicAuth authentication type
+    # In case of tenantId authentication, it is injected as the X-Scope-OrgID header on requests.
+    # In case of basicAuth, it is set as the username.
+    tenant: 'mimir-continuous-test'
+    # -- Password for basicAuth auth (note: can be environment variable from secret attached in extraEnvFrom, e.g. $(PASSWORD))
+    # It should contain an access token created for an access policy that allows metrics reads and writes for the tenant.
```
Comment (dimitarvdimitrov) [inline operations/helm/charts/mimir-distributed/values.yaml:2540]:
```suggestion
    # For GEM, it should contain an access token created for an access policy that allows `metrics:read` and `metrics:write` for the tenant.
```

**Review by dimitarvdimitrov — APPROVED** [review]
LGTM

**Pull request merged**